rotkehlchen/api/v1/encoding.py

```diff
@@ -1176,6 +1185,49 @@ def _validate_blockchain_account_schemas(
             given_addresses.add(address)
 
 
+def _transform_btc_address(
+        ethereum: EthereumManager,
+        given_address: str,
+) -> BTCAddress:
+    """Returns a SegWit/P2PKH/P2SH address (if exist) given an ENS domain.
+
+    NB: ENS domains for BTC store the scriptpubkey. Check EIP-2304.
+    """
+    if not given_address.endswith('.eth'):
+        return BTCAddress(given_address)
+
+    resolved_address = ethereum.ens_lookup(
+        given_address,
+        blockchain=SupportedBlockchain.BITCOIN,
+    )
+    if resolved_address is None:
+        raise ValidationError(
+            f'Given ENS address {given_address} could not be resolved for Bitcoin',
+            field_name='address',
+        ) from None
+
+    address: BTCAddress
```

Comment:
From here and until 1224 it shoudl be moved to its own function in https://github.com/rotki/rotki/blob/develop/rotkehlchen/chain/bitcoin/utils.py

Perhaps scriptpubkey_to_btc_address() should be the name

Reply:
Cool, agree